src/optlang/matrix_interface.py

```diff
@@ -0,0 +1,961 @@
+            )
+        else:
+            P = None
+        q = zeros(nv)
+        q[[vmap[vn] for vn in self.obj_linear_coefs]] = list(
+            self.obj_linear_coefs.values()
+        )
+        q = q * d
+        vbounds = array(
+            [[self.variable_lbs[vn], self.variable_ubs[vn]] for vn in self.variables]
+        )
+
+        if len(self.constraint_coefs) > 0:
+            A = array(
+                [
+                    [cmap[vn[0]], vmap[vn[1]], coef]
+                    for vn, coef in six.iteritems(self.constraint_coefs)
+                ]
+            )
+            bounds = array(
+                [
+                    [self.constraint_lbs[cn], self.constraint_ubs[cn]]
+                    for cn in self.constraints
+                ]
+            )
+            if add_variable_constraints:
+                Av = array([[vmap[k] + nc, vmap[k], 1.0] for k in self.variables])
+                A = concatenate((A, Av))
+                bounds = concatenate((bounds, vbounds))
+            A = csc_matrix(
+                (A[:, 2], (A[:, 0].astype("int64"), A[:, 1].astype("int64"))),
+                shape=(nc + nv, nv),
+            )
+        elif add_variable_constraints:
+            Av = array([[vmap[k], vmap[k], 1.0] for k in self.variables])
+            A = csc_matrix(
+                (Av[:, 2], (Av[:, 0].astype("int64"), Av[:, 1].astype("int64"))),
+                shape=(nv, nv),
+            )
+            bounds = vbounds
+        else:
+            A = None
+            bounds = None
+
+        integer = array([int(vn in self.integer_vars) for vn in self.variables])
+        return P, q, A, bounds, vbounds, integer
+
+    def solve(self):
+        """Solve the problem.
+
+        This will return nothing but will fill the solution attributes.
+
+        Returns
+        -------
+        Nothing.
+        """
+        raise NotImplementedError("This needs to be overwritten by the child class.")
+
+    def reset(self, everything=False):
+        """Reset the public solver solution.
+
+        Parameters
+        ----------
+        everything : bool
+            If true will also clear cached solution bases or warm start solutions.
+
+        Returns
+        -------
+        Nothing.
+        """
+        self.info = None
+        self.primals = {}
+        self.cprimals = {}
+        self.duals = {}
+        self.vduals = {}
+        if everything:
+            self._solution = None
+
+    def still_valid(self, A, bounds):
+        """Check if previous solutions is still feasible.
+
+        Parameters
+        ----------
+        A : numpy.csc_matrix
+            The constraint matrix A.
+        bounds : numpy.array
+            The constraint bounds.
+
+        Returns
+        -------
+        bool
+            Whether the current solution still fulfills the bounds.
+
+        """
+        raise NotImplementedError("This needs to be overwritten by the child class.")
+
+    def clean(self):
```

Comment:
Not sure that I like the name clean. What about `prune` or `reduce`?

Reply:
I like prune.